pkg/cache/controller.go

```diff
@@ -0,0 +1,48 @@
+package cache
+
+import (
+	"fmt"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+)
+
+const (
+	// ClusterIndexName is the name of the index that allows you to filter by cluster
+	ClusterIndexName = "cluster"
+	// ClusterAndNamespaceIndexName is the name of index that allows you to filter by cluster and namespace
+	ClusterAndNamespaceIndexName = "cluster-and-namespace"
+)
+
+// ClusterIndexFunc indexes by cluster name
+func ClusterIndexFunc(obj interface{}) ([]string, error) {
+	meta, err := meta.Accessor(obj)
+	if err != nil {
+		return []string{""}, fmt.Errorf("object has no meta: %v", err)
+	}
+	return []string{meta.GetZZZ_DeprecatedClusterName()}, nil
```

Comment:
```suggestion
	return []string{logicalcluster.From(meta).String()}, nil
```

Reply:
Global comment

Reply:
hmm, as far as I can see there's no equivalent for inserting one into the metadata, so I'd still be referring to the field by hand in the lister implementation. Changed here (and updated the others to refer to ClusterName instead of ZZZ_DeprecatedClusterName) to match the implementation in logicalcluster.From

Reply:
That's fine for now, thanks. We may think about switching to calling the key func and using ByIndex instead of constructing an ObjectMeta